### PR TITLE
Remove GRPC Builder usages in TaskExecutorManager

### PR DESCRIPTION
**Pull request by bradyoo**
Addresses parts of https://github.com/Alluxio/alluxio/issues/10175

### CODE REVIEW

**Comment by bradyoo** [comment]
jenkins, test this please

**Comment by bradyoo** [comment]
Jenkins, test this please

**Comment by bradyoo** [comment]
Jenkins, test this please.

**Comment by bradyoo** [comment]
jenkins, test this please.

**Comment by bradyoo** [comment]
jenkins, test this please

**Comment by bradyoo** [comment]
@calvinjia: Can you look at this? This has been around for a while and a lot of new code I'm trying to write conflicts with this.

File: job/common/src/main/java/alluxio/job/wire/TaskInfo.java [hunk 1]
```diff
@@ -137,18 +154,24 @@ public TaskInfo setErrorMessage(String errorMessage) {
...
-    ByteBuffer result =
-        mResult == null ? null : ByteBuffer.wrap(SerializationUtils.serialize(mResult));
+  public alluxio.grpc.TaskInfo toProto() {
+    ByteBuffer result = null;
+    try {
+      result = mResult == null ? null : ByteBuffer.wrap(SerializationUtils.serialize(mResult));
+    } catch (IOException e) {
+      // TODO(bradley) better error handling
+      LOG.warn("Failed to serialize {} : {}", mResult, e.getMessage());
+      LOG.debug("Exception: ", e);
```
Comment (calvinjia) [inline job/common/src/main/java/alluxio/job/wire/TaskInfo.java]: I suggest using `LOG.isDebugEnabled` and only log one message, it is not guaranteed that these two messages will be printed together.

File: job/server/src/main/java/alluxio/worker/job/task/TaskExecutorManager.java [hunk 2]
```diff
@@ -71,20 +68,13 @@ public TaskExecutorManager(int taskExecutorPoolSize) {
    * @param taskId the task id
    * @param result the task execution result
    */
-  public synchronized void notifyTaskCompletion(long jobId, int taskId, Object result) {
+  public synchronized void notifyTaskCompletion(long jobId, int taskId, Serializable result) {
```
Comment (calvinjia) [inline job/server/src/main/java/alluxio/worker/job/task/TaskExecutorManager.java]: Do any callers need to be updated?

File: job/server/src/main/java/alluxio/worker/job/task/TaskExecutorManager.java [hunk 3]
```diff
@@ -132,12 +122,11 @@ public synchronized void executeTask(long jobId, int taskId, JobConfig jobConfig
         .submit(new TaskExecutor(jobId, taskId, jobConfig, taskArgs, context, this));
     Pair<Long, Integer> id = new Pair<>(jobId, taskId);
     mTaskFutures.put(id, future);
-    TaskInfo.Builder taskInfo = TaskInfo.newBuilder();
-    taskInfo.setJobId(jobId);
-    taskInfo.setTaskId(taskId);
-    taskInfo.setStatus(Status.RUNNING);
+    TaskInfo taskInfo = new alluxio.job.wire.TaskInfo(jobId, taskId,
+        Status.RUNNING);
```
Comment (calvinjia) [inline job/server/src/main/java/alluxio/worker/job/task/TaskExecutorManager.java]: Can this be on one line?

File: job/common/src/main/java/alluxio/job/wire/TaskInfo.java (hunk 1 above)
Comment (bradyoo) [reply job/common/src/main/java/alluxio/job/wire/TaskInfo.java]: This is actually a copy paste from an existing code in TaskExecutorManager.java

File: job/server/src/main/java/alluxio/worker/job/task/TaskExecutorManager.java (hunk 2 above)
Comment (bradyoo) [reply job/server/src/main/java/alluxio/worker/job/task/TaskExecutorManager.java]: Nope. It was already passing in a Serializable

**Comment by bradyoo** [comment]
This has now been rebased on top of the large refactor PR.

**Pull request closed without merging**